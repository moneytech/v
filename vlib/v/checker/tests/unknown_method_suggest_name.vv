import hash.crc32

struct Point {
	x   int
	y   int
	z   int
	ccc crc32.Crc33
}

struct Vector {
	x int
	y int
	z int
}

fn (p Point) translate(v Vector) Point {
	return Point{p.x + v.x, p.y + v.y, p.z + v.z}
}

fn (p Point) identity() Point {
	return Point{1, 1, 1}
}

fn main() {
	p := Point{1, 2, 3}
	v := Vector{5, 5, 10}
	z := p.tranzlate(v)
	println('p: $p')
	println('v: $v')
	println('z: $z')
}
