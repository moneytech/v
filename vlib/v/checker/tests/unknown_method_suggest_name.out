vlib/v/checker/tests/unknown_method_suggest_name.v:7:2: error: unknown type `hash.crc32.Crc33`. Did you mean `crc32.Crc32` ?
    5 |     y   int
    6 |     z   int
    7 |     ccc crc32.Crc33
      |     ~~~~~~~~~~~~~~~
    8 | }
    9 |
vlib/v/checker/tests/unknown_method_suggest_name.v:27:9: error: unknown method: `Point.tranzlate`. Did you mean `translate` ?
   25 |     p := Point{1, 2, 3}
   26 |     v := Vector{5, 5, 10}
   27 |     z := p.tranzlate(v)
      |            ~~~~~~~~~~~~
   28 |     println('p: $p')
   29 |     println('v: $v')
